### ISSUE TITLE
Mismatched input / output parameters in B2CContactResolve flow

### ISSUE THREAD

**Issue by KennethRobbeFrucon** [sfb2csolutionarchitects/b2c-crm-sync#125]
I was able to get the solution up and running (not quite in 15 minutes, but I'm getting there :) ) and I must say I'm quite impressed!

However, while playing around with it, I got stuck trying to do customer resolution for an anonymous user (e.g. just based on an customer list ID, email address and last name). To do this, I called the `app.customer.retrieve` hook in B2C Commerce, which in turn calls the B2CContactResolve-flow through a REST service call. Is this the correct approach to take?

The flow results in an error for me when the Apex action is invoked:
```
B2CIACUSTOMERRESOLUTION (APEX): B2CCustomerResolution_Action
Inputs:
contactListList = {!ContactList} ([Contact (0031X00000hRXeBQAW)])
Error Occurred: An invalid input parameter was specified for Apex action B2CIACustomerResolution: contactListList
```

It seems like the input / output parameters for that Apex action are mismatched. Both 'contactListList' and 'output' are reported as missing, the Apex action expects and exposes 'contactList':
![Screenshot_6](https://user-images.githubusercontent.com/34708318/121693027-053c3080-cac9-11eb-8860-bf0bb06b493e.png)

In addition to that, the variable ContactList in the flow is defined as a record collection, but the apex action expects a single record (the invokable action puts it in a list anyways). Due to this, the request body for the service call from B2C (getRetrieveRequestBody in src\sfcc\cartridges\int_b2ccrmsync\cartridge\scripts\models\customer.js) is also different, ContactList shouldn't be an array.

**Comment by b2csolutionarchitect** [comment]
Hi @KennethRobbeFrucon -- thank you for your interest and efforts working through the b2c-crm-sync setup.  So, the `RESOLVE` REST API is a bit of a work in progress.  We don't recommend using it just yet -- as it needs to be aligned with the `PROCESS` REST API.  Right now, they don't leverage the same resolution logic.  

> Please Don't use the `RESOLVE` API until we resolve #38.

You should (for now) always use the `PROCESS` REST API to resolve a customer and get a Contact ID.  The `PROCESS` API will be the only API that generates a Contact ID for you if your customer can't be resolved AND return a resolved Contact ID.  So for now -- always use the `~/cartridge/scripts/hooks/customer.process` hook.  It will do the resolution you need -- AND return an existing or new Account ID / Contact ID.

Thanks again for your time and interest.  Please reach out if you have any other questions -- ok?  Super happy to support you.

**Comment by KennethRobbeFrucon** [comment]
Thanks for your quick feedback! Happy to help test this, it does seem very promising.

I started looking into the `customer.retrieve` hook to e.g. do resolution of unregistered customers at checkout because the `customer.process` hook currently only takes an actual customer profile (at the side of B2C).

Thanks again for your support, let me know if I can assist anywhere!

**Comment by b2csolutionarchitect** [comment]
@KennethRobbeFrucon -- if you have cycles, we could ABSOLUTELY use some help with #38.  Let me know if you're interested, ok?  I can write-up requirements and success criteria in the ticket for you.  Basically, the effort for #38 is to enable the use-case you originally reached-out about.

- We want to have a Read-Only service that accepts customer attributes -- and provides resolution results
- It will provide precise results (either exact matches or multiple matches based on what's available)
- It will not create customer profiles -- it only exposes Account / Contact pairs based on provided criteria

> The version of the resolutionEngine that's attached to this specific flow is fairly legacy -- the work would involve leveraging what we're using in the `B2CContactProcess` flow (from a resolution standpoint) and exposing all the results.  

I'm happy to discuss in more detail -- so let me know what your thoughts are.  I'll make sure to send you a hoody too ;)

**Comment by KennethRobbeFrucon** [comment]
@b2csolutionarchitect I'm surely interested, but I'll have to check internally how much time I can allocate to this and what use cases we actually want to support through the connector. Let me get back to you!

**Comment by b2csolutionarchitect** [comment]
Thanks @KennethRobbeFrucon -- I'm definitely interested in hearing more about the use-cases that you're interested in and what you're trying to do with B2C Commerce.  Thanks in advance, and looking forward to hearing from you.

**Comment by b2csolutionarchitect** [comment]
We've gone ahead and re-architected the B2CContactResolve flow so that it accepts the same inputs as B2CContactProcess -- and returns the contact that was found, or the duplicates that were resolved.  The unit tests for this service and custom hook in B2C Commerce are also aligned to the new service definition.  This has all been merged into the [v0.9.0 release-branch](https://github.com/sfb2csolutionarchitects/b2c-crm-sync/tree/release/0.9.0).
